pkgs/servers/sql/postgresql/ext/pgrouting.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, postgresql, perl, cmake, boost, gmp, cgal, mpfr }:
+
+stdenv.mkDerivation rec {
+  pname = "pgrouting";
+  version = "2.6.2";
+
+  nativeBuildInputs = [ cmake perl ];
+  buildInputs = [ postgresql boost gmp cgal mpfr ];
+
+  src = fetchFromGitHub {
+    owner  = "pgRouting";
+    repo   = pname;
+    rev    = "refs/tags/v${version}";
+    sha256 = "09xy5pmiwq0lxf2m8p4q5r892mfmn32vf8m75p84jnz4707z1l0j";
+  };
+
+  installPhase = ''
```

Comment:
you need to create an empty `$out/bin`, because the current PostgreSQL extension mechanism requires a bin directory, see https://github.com/NixOS/nixpkgs/issues/22653

Reply:
Hmm.. seems that should be solved in a more generic way to avoid patching each pg extension.

But I'll add the change anyway since other pg extensions seems to be doing this as well.

Reply:
This change somehow broke my ability to switch pg versions with `sudo nixos-rebuild test`, now I get these errors:
```
rm: cannot remove '/nix/store/8bxbi71wwpm205bd34l9xhx1vv39kzg7-postgresql-and-plugins-10.6/bin/pg_config': Permission denied
rm: cannot remove '/nix/store/8bxbi71wwpm205bd34l9xhx1vv39kzg7-postgresql-and-plugins-10.6/bin/postgres': Permission denied
rm: cannot remove '/nix/store/8bxbi71wwpm205bd34l9xhx1vv39kzg7-postgresql-and-plugins-10.6/bin/pg_ctl': Permission denied
```

Reply:
I think it fails because is in the `postInstall` phase, can you move it to `installPhase`, LGTM otherwise. Before I merge it would you mind squashing the commits?

Reply:
Done!